localization/graph_localizer/src/serialization.cc

```diff
@@ -0,0 +1,82 @@
+/* Copyright (c) 2017, United States Government, as represented by the
+ * Administrator of the National Aeronautics and Space Administration.
+ *
+ * All rights reserved.
+ *
+ * The Astrobee platform is licensed under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <graph_localizer/serialization.h>
+
+#include <gtsam/base/serialization.h>
+
+#include <boost/serialization/serialization.hpp>
+
+// Value types used in GraphValues
+#include <gtsam/base/Vector.h>
+#include <gtsam/geometry/Cal3_S2.h>
+#include <gtsam/geometry/Point3.h>
+#include <gtsam/geometry/Pose3.h>
+#include <gtsam/inference/Symbol.h>
+#include <gtsam/navigation/ImuBias.h>
+#include <gtsam/navigation/NavState.h>
+
+// Factors used in NonlinearFactorGraph
+#include <graph_localizer/loc_projection_factor.h>
+#include <graph_localizer/robust_smart_projection_pose_factor.h>
+#include <gtsam/nonlinear/NonlinearFactor.h>
+#include <gtsam/slam/PriorFactor.h>
+#include <gtsam/slam/SmartProjectionPoseFactor.h>
+
+// Export all classes derived from Value so they can
+// be properly serialized using base class (value) pointers in GraphValues
+GTSAM_VALUE_EXPORT(gtsam::Cal3_S2);
```

Comment:
What is the serialization for? Just curious

Reply:
This is for visualization, we can serialize the graph and deserialize it in a localization plugin and access all of its member functions and data.  This makes it easy to add visualization tools rather than having to populate a custom ros msg and add fields every time new data is desired.